pkg/cmd/secret/list/list.go

```diff
@@ -114,9 +117,12 @@ func listRun(opts *ListOptions) error {
 		return fmt.Errorf("%s secrets are not supported for %s", secretEntity, secretApp)
 	}
 
-	var secrets []Secret
 	showSelectedRepoInfo := opts.IO.IsStdoutTTY()
+	if !showSelectedRepoInfo && opts.Exporter != nil {
+		showSelectedRepoInfo = slices.Contains(opts.Exporter.Fields(), secretFieldNumSelectedRepos)
+	}
```

Comment:
Can this be simplified like so?

```suggestion
	showSelectedRepoInfo := opts.IO.IsStdoutTTY() || opts.Exporter != nil
```

`numSelectedRepos` will always be in the exported fields list, so it is really a question of whether standard out is TTY or we're exporting the fields, right?

Reply:
It actually depends on the argument. For example the caller may just run `gh secret list --json name`. In this case, we don't need the extra requests to populate `numSelectedRepos`.

Reply:
Good point, thanks for clarifying ❤️

Reply:
No worries. Actually, I'm thinking that we can go a step further and even avoid populating the `numSelectedRepos` in the interactive mode, **only if** there's a `--json` argument without the `numSelectedRepos` fields in it. How about that?